recipe/build.sh

```diff
@@ -67,13 +67,15 @@ fi
     debug-symbols=off \
     threading=multi \
     runtime-link=shared \
-    link=static,shared \
+    link=shared \
     toolset=${TOOLSET} \
-    python="${PY_VER}" \
     include="${INCLUDE_PATH}" \
     cxxflags="${CXXFLAGS}" \
     linkflags="${LINKFLAGS}" \
     --layout=system \
-    --with-python \
     -j"${CPU_COUNT}" \
-    install 2>&1 | tee b2.log
+    install
+
+# python headers packaged in build-py.sh (because we need to build per python version)
```

Comment:
Are you sure about this?

Reply:
Perhaps "need" is the wrong word. Historically, this has been packaged per python version, which also ends up reflected in the library name (e.g. `libboost_python39.so`). I kept that setup, since I wanted to minimise changes while merging the feedstocks, and because I don't know if it would actually be possible to do just once (which would be nice of course!)

Reply:
Let's keep all the headers at the same place. We don't have to keep carrying tech debt forever.

Reply:
> We don't have to keep carrying tech debt forever.

I don't have all the background on this, but I presume the headers for `libboost-python` are not actually dependent on the python version, only the compiled libraries. So - IIUC - we still have to compile `libboost-python` per python version, but we can ensure that the python headers are already present in `libboost-headers`.

That sounds like a clear improvement indeed, and happy to give it a shot.